DataFormats/TrackReco/src/HitPattern.cc

```diff
@@ -323,30 +323,21 @@ uint16_t HitPattern::getHitPatternByAbsoluteIndex(int position) const
     }
 }
 
-bool HitPattern::hasValidHitInFirstPixelBarrel() const
-{
-    for (int i = beginTrackHits; i < endTrackHits; ++i) {
-        uint16_t pattern = getHitPatternByAbsoluteIndex(i);
-        if (pixelBarrelHitFilter(pattern) && (getLayer(pattern) == 1)
-                && validHitFilter(pattern)) {
-            return true;
-        }
-    }
-    return false;
-}
 
-bool HitPattern::hasValidHitInFirstPixelEndcap() const
```

Comment:
No forget about
Please keep the old implementation w/o any further possible performace impact.
If you need a specific implementation make a new method.
Do not modify what was carefully crafted.